### ISSUE TITLE
Detect the same `if` conditions

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#2241]
Today I saw this code:

```python
if isinstance(x, int):
    ...
elif isinstance(x, str):
    ...
elif isinstance(x, int): ...
```

The thing is that `isinstance(x, int)` is duplicated. And the second branch won't ever be called.

We need to detecct this kind of problems and report them.

**Comment by Dreamsorcerer** [comment]
Isn't this a case for mypy? It should warn that the 3rd statement is unreachable (if you've enabled that option).

**Comment by sobolevn** [comment]
Yes, mypy can catch some cases. But, not this one for example:

```python
def some(x: int) -> None:
    if x == 1:
        ...
    elif x == 2:
        ...
    elif x == 1:
        ...
```

**Comment by sobolevn** [comment]
This is a different issue, I guess 🙂 
We can detect similar conditions that go as the next statement.
